iot_config/src/main.rs

```diff
@@ -0,0 +1,98 @@
+use anyhow::{Error, Result};
+use clap::Parser;
+use futures_util::TryFutureExt;
+use helium_proto::services::iot_config::{
+    GatewayServer, OrgServer, RouteServer, SessionKeyFilterServer,
+};
+use iot_config::{
+    gateway_service::GatewayService, org_service::OrgService, route_service::RouteService,
+    session_key_service::SessionKeyFilterService, settings::Settings,
+};
+use std::path::PathBuf;
+use tokio::signal;
+use tonic::transport;
+use tracing_subscriber::{layer::SubscriberExt, util::SubscriberInitExt};
+
+#[derive(Debug, clap::Parser)]
+#[clap(version = env!("CARGO_PKG_VERSION"))]
+#[clap(about = "Helium IoT Config Service")]
+pub struct Cli {
+    /// Optional configuration file to use. If present, the toml file at the
+    /// given path will be loaded. Environment variables can override the
+    /// settings in the given file.
+    #[clap(short = 'c')]
+    config: Option<PathBuf>,
+
+    #[clap(subcommand)]
+    cmd: Cmd,
+}
+
+impl Cli {
+    pub async fn run(self) -> Result<()> {
+        let settings = Settings::new(self.config)?;
+        self.cmd.run(settings).await
+    }
+}
+
+#[derive(Debug, clap::Subcommand)]
+pub enum Cmd {
+    Server(Daemon),
+}
+
+impl Cmd {
+    pub async fn run(&self, settings: Settings) -> Result<()> {
+        match self {
+            Self::Server(cmd) => cmd.run(&settings).await,
+        }
+    }
+}
+
+#[derive(Debug, clap::Args)]
+pub struct Daemon {}
```

Comment:
FYI, field less structs do not need the `{}`:
```rust
pub struct Daemon;
```

Reply:
good to know; surprised that's not something clippy would complain about